### PR TITLE
Add readme for spec

### PR DESCRIPTION
**Pull request by iamniting**
As part of the PR initial readme is added and ignored readme file for jscpd

### CODE REVIEW

**Review by Madhu-1 — COMMENTED** [review]
content LGTM. please correct the commit message and fix CI issue

as a follow up we need to document below items

* Document Error codes
* More detailed explanation of each operation.

File: README.md [hunk 1]
```diff
@@ -1 +1,218 @@
...
+
+
+## Architecture
+
+![](images/arch.png)
+
+
+## RPC Interface
+
+* **Controller Service**: The Controller Plugin MUST implement this sets of RPCs.
```
Comment (sp98) [inline README.md:48]:
```suggestion
* **Controller Service**: The Controller Plugin MUST implement these sets of RPCs.
```

File: README.md [hunk 2]
```diff
@@ -1 +1,218 @@
...
+
+
+```protobuf
+// Controller holds the RPC Methods for replication and all the methods it
```
Comment (sp98) [inline README.md:52]:
```suggestion
// Controller holds the RPC methods for replication and all the methods it
```

File: README.md [hunk 3]
```diff
@@ -1 +1,218 @@
...
+message DisableVolumeReplicationResponse {
+}
+```
+
+
+#### PromoteVolume
+
+```protobuf
+// PromoteVolumeRequest holds the required information to promote volume as a
+// primary on local cluster.
```
Comment (sp98) [inline README.md:132]: local cluster ?

**Review by sp98 — COMMENTED** [review]
NIT

File: README.md (hunk 3 above)
Comment (iamniting) [reply README.md:132]: local cluster means the second cluster where it is not primary and should be run from that itself locally.

File: README.md [hunk 4]
```diff
@@ -1 +1,218 @@
...
+## Objective
+
+Define a standard that will enable storage vendors (SP) to develop controllers/plugins for DR or to talk to the different CO systems.
+
+
+### Goals in MVP
+
+The New standard will
+* Provide API at volume level granularity.
+* Enable SP authors to write one replication compliant Plugin that “just works” across all COs that implement RPC.
```
Comment (sp98) [inline README.md:23]:
```suggestion
* Enable SP authors to write one replication compliant plugin that “just works” across all COs that implement RPC.
```

File: README.md [hunk 5]
```diff
@@ -1 +1,218 @@
...
+
+
+## Objective
+
+Define a standard that will enable storage vendors (SP) to develop controllers/plugins for DR or to talk to the different CO systems.
+
+
+### Goals in MVP
+
+The New standard will
```
Comment (sp98) [inline README.md:21]:
```suggestion
The new standard will
```

File: README.md [hunk 6]
```diff
@@ -1 +1,218 @@
-# spec
\ No newline at end of file
+# spec
+
+## Terminology
+
+| Term | Definition |
+|-------------------|--------------------------------------------------|
+| VolumeID | The Identifier of the volume generated by the Plugin. |
```
Comment (sp98) [inline README.md:7]:
```suggestion
| VolumeID | The identifier of the volume generated by the plugin. |
```

**Review by sp98 — COMMENTED** [review]
just a few more nits

File: README.md (hunk 3 above)
Comment (sp98) [reply README.md:132]: got it.

**Review by Madhu-1 — APPROVED** [review]

**Comment by Madhu-1** [comment]
@sp98 PTAL

**Review by sp98 — APPROVED** [review]

**Pull request merged**